registry/api/errcode/errors.go

```diff
@@ -69,20 +69,27 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // WithDetail creates a new Error struct based on the passed-in info and
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
-	if err, ok := detail.(error); ok {
-		detail = err.Error()
-	}
+	return Error{
+		Code:    ec,
+		Message: ec.Message(),
+	}.AddDetail(detail)
+}
 
+// WithArgs creates a new Error struct and sets the Args slice
+func (ec ErrorCode) WithArgs(args ...interface{}) Error {
 	return Error{
-		Code:   ec,
-		Detail: detail,
-	}
+		Code:    ec,
+		Message: ec.Message(),
+	}.AddArgs(args...)
 }
 
 // Error provides a wrapper around ErrorCode with extra Details provided.
+// TODO(duglin): See if we need an "args" property so we can do the
```

Comment:
Place these kinds of TODO message in teh struct block. No need to have them show up in godoc:

```
type MyType struct {
    A string

    // TODO(duglin): See if we need an "args" property so we can do the...
}
```